app/models/Case.scala

```diff
@@ -35,11 +35,12 @@ case class Case
   attachments: Seq[Attachment] = Seq.empty
 ) {
 
-  def hasEoriNumber(eoriNumber: String): Boolean = {
-    application.holder.eori == eoriNumber || application.agent.exists(_.eoriDetails.eori == eoriNumber)
-  }
+  def hasEoriNumber(eoriNumber: String): Boolean = application.holder.eori == eoriNumber || application.agent.exists(_.eoriDetails.eori == eoriNumber)
 
   def hasActiveDecision: Boolean = this.decision.flatMap(_.effectiveEndDate).exists(_.compareTo(Instant.now) >= 0)
 
   def hasExpiredDecision: Boolean = this.decision.flatMap(_.effectiveEndDate).exists(_.compareTo(Instant.now) < 0)
+
+  def hasRuling: Boolean = decision.exists(d => d.effectiveStartDate.isDefined && d.effectiveEndDate.isDefined)
```

Comment:
is this the proper way to make sure there is a ruling ?

Reply:
Not sure if that is the "complete" definition of "case has a ruling".    It's good enough (IMO) to ensure that a ruling cert pdf will have all of the information it requires (i.e. a Decision block with start and end dates).

The only other additional check I can think of is that the case status is Completed or Canceled.  I'll check with Matt/Joe

Reply:
Those are my acceptance criteria in DIT-79 about what to show as a Ruling : 

> Ruling status mappings must be presented as follows:
> "Active" where Case.Status = Complete AND Current Date <= ruling End Date
> "Expired" where Case.Status = Complete and Current Date > ruling End Date
> "Cancelled" where Case.Status = Cancelled

Reply:
Thanks - I've added a question on my ticket - https://jira.tools.tax.service.gov.uk/browse/DIT-81

Reply:
@SteveSugden 
as agreed with @pllaurado , we would like this method to assert the case status (you can keep the dates assertion, but perhaps better to get rid of it and just check the case status == completed / decision_made)

Reply:
Ok.  But it will have to be status = Completed or Cancelled